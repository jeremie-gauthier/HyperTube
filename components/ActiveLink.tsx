import { useRouter } from "next/router";
import classNames from "classnames";
import Link from "next/link";
import styles from "./ActiveLink.module.scss";

type ActiveLinkProps = {
  children: React.ReactNode;
  href: string;
};

function ActiveLink({ children, href }: ActiveLinkProps): JSX.Element {
  const router = useRouter();
  const isActive = router.pathname === href;
  const activeLink = classNames({
    [styles.inactive]: !isActive,
    [styles.active]: isActive,
  });

  return (
    <Link href={href}>
      <div className={activeLink}>
        {children}
        <div className={styles.border} />
      </div>
    </Link>
  );
}

export default ActiveLink;
