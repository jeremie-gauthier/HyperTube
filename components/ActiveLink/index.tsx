import { useRouter } from "next/router";
import classNames from "classnames";
import Link from "next/link";
import styles from "./ActiveLink.module.scss";

type ActiveLinkProps = {
  children: React.ReactNode;
  href: string;
} & Partial<DefaultProps>;

type DefaultProps = {
  className: string;
  inactiveClassName: string;
  activeClassName: string;
};

function ActiveLink({
  children,
  href,
  className = "",
  inactiveClassName = "",
  activeClassName = "",
  ...rest
}: ActiveLinkProps): JSX.Element {
  const router = useRouter();
  const isActive = router.pathname === href;
  const activeLink = classNames({
    [className]: true,
    [`${styles.inactive} ${inactiveClassName}`]: !isActive,
    [`${styles.active} ${activeClassName}`]: isActive,
  });

  return (
    <Link href={href}>
      <a className={activeLink} {...rest}>
        {children}
      </a>
    </Link>
  );
}

export default ActiveLink;
