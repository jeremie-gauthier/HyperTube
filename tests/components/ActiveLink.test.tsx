import React from "react";
import ActiveLink from "@/components/ActiveLink";
import renderer from "react-test-renderer";

describe("ActiveLink", () => {
  test("<ActiveLink /> should render correctly when active", () => {
    const tree = renderer
      .create(<ActiveLink href="/">Home</ActiveLink>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("<ActiveLink /> should render correctly when inactive", () => {
    const tree = renderer
      .create(<ActiveLink href="/movies">Home</ActiveLink>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
