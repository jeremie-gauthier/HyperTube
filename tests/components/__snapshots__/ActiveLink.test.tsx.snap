// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveLink <ActiveLink /> should render correctly 1`] = `
<a
  className=" inactive "
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Home
</a>
`;
